app/index.js

```diff
@@ -0,0 +1,153 @@
+		// Calling the super constructor is important so our generator is correctly set up
+		super(args, opts);
+		this.argument('jsonFileNames', { type: String, required: false });
+		this.jsonFileNames = this.options.jsonFileNames || '';
+		this.jsonFileNames = this.jsonFileNames.split(',');
+	}
+
+	createCrudModules(){
+		if(!this.jsonFileNames.length){
+			console.log(chalk.bold.red('Please provide json file.'));
+		} else {
+			this.jsonFileNames.forEach((jsonFileName)=>{
+				let moduleName = jsonFileName.replace('.json','');
+				moduleName = s.camelize(s.slugify(s.humanize(moduleName)));
+				this._createCrudModule(moduleName,jsonFileName);
+			})
+		}
+	}
+
+	_createCrudModule(moduleName,jsonFileName){
+		//validate file
+		if(!moduleName){
+			console.log(chalk.bold.red('Please provide module name.'));
+			return;
+		}
+		if(!this.fs.exists(jsonFileName)){
+			console.log(chalk.bold.red(`${jsonFileName} file does not exist.`));
+			return;
+		}
+		let jsonContent = this.fs.readJSON(jsonFileName);
+		if(!_.get(jsonContent,'response.data.length')){
+			console.log(chalk.bold.red('Please provide valid JSON file.'));
+			return;
+		}
+		let fields = _.get(jsonContent,'response.data[0]');
+		if(!_.isPlainObject(fields)){
+			console.log(chalk.bold.red('Please provide valid JSON file.'));
+			return;
+		}
+		let fieldKeys = _.keys(fields);
+		fields = {};
+		_.each(fieldKeys,(field)=>fields[field] = {title:field,type:'string'});
+		fields = JSON.stringify(fields,null,4);
+		fields = _.replace(fields,new RegExp('\n','g'),"\n\t\t");
+
+		//Add component file
+		let moduleNameSingular = moduleName, //singular crud name
+			moduleNamePlural = moduleName+'s', //plural crud name
+			ModuleNameSingular = s.capitalize(moduleName),   //singular crud name capital case
+			ModuleNamePlural =  s.capitalize(moduleName)+'s', //plural crud name capital case
+			componentName = ModuleNameSingular+'Component',
+			serviceName =  ModuleNameSingular+'Service',
+			replaceVariables ={ ModuleNameSingular,ModuleNamePlural ,moduleNamePlural,moduleNameSingular,fields},
+			componentHtml = `${moduleNamePlural}/${moduleNameSingular}.component.html`,
+			componentTs = `${moduleNamePlural}/${moduleNameSingular}.component.ts`,
+			componentService = `${moduleNamePlural}/${moduleNameSingular}.service.ts`;
+		this.fs.copyTpl(
+			this.templatePath('crud-module/crud.component.html'),
+			this.destinationPath(`src/app/pages/tables/${componentHtml}`),
+			replaceVariables
+		);
+		this.fs.copyTpl(
+			this.templatePath('crud-module/crud.component.ts'),
+			this.destinationPath(`src/app/pages/tables/${componentTs}`),
+			replaceVariables
+		);
+		this.fs.copyTpl(
+			this.templatePath('crud-module/crud.service.ts'),
+			this.destinationPath(`src/app/pages/tables/${componentService}`),
+			replaceVariables
+		);
+		this.fs.copyTpl(
+			this.templatePath('table-paging.component.ts'),
+			this.destinationPath(`src/app/pages/tables/table-paging.component.ts`),
```

Comment:
Nope currently server-side paging is not supported in the table component, so I have to create a custom component which can manage the server side paging so it would not be dynamic.